protocol-units/da/m1/util/src/config/local/m1_da_light_node.rs

```diff
@@ -5,7 +5,88 @@ use crate::config::common::{
 	default_m1_da_light_node_connection_port, default_m1_da_light_node_listen_hostname,
 	default_m1_da_light_node_listen_port,
 };
+use ecdsa::SigningKey;
+use k256::Secp256k1;
 use serde::{Deserialize, Serialize};
+use std::collections::HashSet;
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+pub struct DaSigners {
+	pub private_key_hex: String,
+	pub public_keys_hex: HashSet<String>,
+}
+
+/// The default da signing private key
+pub fn default_da_signing_private_key() -> SigningKey<Secp256k1> {
+	match std::env::var("DA_SIGNING_PRIVATE_KEY") {
+		Ok(val) => {
+			// decode from hex to bytes 32
+			let hex_bytes = hex::decode(val).expect("Invalid hex string");
+
+			// todo: maybe remove the unwrap and catch for a random signing key
+			let signing_key_bytes: &[u8; 32] =
+				hex_bytes.as_slice().try_into().expect("Slice with incorrect length");
+			SigningKey::from_bytes(signing_key_bytes.into()).unwrap()
+		}
+		Err(std::env::VarError::NotPresent) => SigningKey::random(
+			// rand_core
+			&mut rand::rngs::OsRng,
+		),
+		Err(_) => panic!("Invalid DA_SIGNING_PRIVATE_KEY"),
+	}
+}
+
+pub fn default_da_signers_sec1_keys() -> HashSet<String> {
+	match std::env::var("DA_SIGNERS_SEC1_KEYS") {
+		Ok(val) => val.split(',').map(|s| s.to_string()).collect(),
+		Err(std::env::VarError::NotPresent) => HashSet::new(),
+		Err(_) => panic!("Invalid DA_SIGNERS_SEC1_KEYS"),
+	}
+}
+
+pub fn default_da_signers() -> DaSigners {
+	let da_signer = default_da_signing_private_key();
+
+	// always trust yourself
+	let mut trusted_signers = HashSet::new();
+	let sec1_hex = hex::encode(da_signer.verifying_key().to_sec1_bytes().to_vec());
+	trusted_signers.insert(sec1_hex);
+
+	// add the other specified signers
+	let additional_signers = default_da_signers_sec1_keys();
+	trusted_signers.extend(additional_signers);
+
+	DaSigners {
+		private_key_hex: hex::encode(da_signer.to_bytes().as_slice()),
+		public_keys_hex: trusted_signers,
+	}
+}
+
+#[cfg(test)]
+pub mod signers_serialization_test {
+
+	use super::*;
+
+	#[test]
+	fn test_signing_key() -> Result<(), anyhow::Error> {
```

Comment:
This only tests the byte format of the signing key, why do we need the test for dependency functionality?